### PR TITLE
Allow to configure extra nginx conf in $app.d/ directory

### PR DESCRIPTION
**Pull request by fflorent**
## Problem

- As a user, I want to use my_webapp with a simple HTTP authentication, and cannot keep extra nginx configuration accross updates

## Solution

- Create a `/etc/nginx/conf.d/$domain.d/$app.d` directory
- Allow the user to put any configuration file in it which will be loaded if the configuration file ends with `.conf` extension
- Use this directory for loading or removing the PHP directives for nginx

## PR Status

- [x] Code finished and ready to be reviewed/tested
- [x] The fix/enhancement were manually tested (if applicable)

## Automatic tests

Automatic tests can be triggered on https://ci-apps-dev.yunohost.org/ *after creating the PR*, by commenting "!testme", "!gogogadgetoci" or "By the power of systemd, I invoke The Great App CI to test this Pull Request!". (N.B. : for this to work you need to be a member of the Yunohost-Apps organization)

### CODE REVIEW

File: scripts/restore [hunk 1]
```diff
@@ -58,6 +58,7 @@ fi
 #=================================================
 
 ynh_restore_file --origin_path="/etc/nginx/conf.d/$domain.d/$app.conf"
+ynh_restore_file --origin_path="/etc/nginx/conf.d/$domain.d/$app.d/"
```
Comment (fflorent) [inline scripts/restore:61]: This causes a warning due to these lines in the helpers:

https://github.com/YunoHost/yunohost/blob/394907ff0df218ec926f43c2e2ba006c68cc3f8a/helpers/backup#L292-L295

**Comment by fflorent** [comment]
!testme

**Comment by fflorent** [comment]
[](https://ci-apps-dev.yunohost.org/ci/job/5035)

**Comment by fflorent** [comment]
The cause of the failure is this:
```

 ! General stuff, misc helper usage

    - READMEs are to be automatically generated using https://github.com/YunoHost/apps/tree/master/tools/README-generator.
    - You are encouraged to create a doc/DISCLAIMER.md file, which should contain any important information to be presented to the admin before installation. Check https://github.com/YunoHost/example_ynh/blob/master/doc/DISCLAIMER.md for more details (it should be somewhat equivalent to the old 'Known limitations' and 'Specific features' section). (It's not mandatory to create this file if you're absolutely sure there's no relevant info to show to the user)
    - If relevant for this app, screenshots can be added in a doc/screenshots/ folder.
    ✘ It looks like the README was not generated automatically by https://github.com/YunoHost/apps/tree/master/tools/README-generator. Note that nowadays you are not suppose to edit README.md, the yunohost bot will usually automatically update it if your app is hosted in the YunoHost-Apps org ... or you can also generate it by running the README-generator yourself.

```

However, I didn't change the README file. Or did my patch made the check fail somehow?

**Comment by fflorent** [comment]
!testme

**Comment by fflorent** [comment]
Run package_check locally for a5d3f51aa5253ce7f96cd0f419bbd312e86761ed, I got this:
```
 ============================================
 Tests summary
 ============================================


------------- Test complet -------------
Package linter:                OK  (3 possible improvements)
Install (root):                OK
Install (subdir):              OK
Install (private):             OK
Install (multi):               OK
Backup/restore:                OK
Upgrade:                       OK
Upgrade (a050e555):            OK
Change url:                    OK
------------- Test without sftp -------------
Install (subdir):              OK
Backup/restore:                OK
Upgrade:                       OK
------------- Test without mysql -------------
Install (subdir):              OK
Backup/restore:                OK
Upgrade:                       OK
------------- Test without PHP -------------
Install (subdir):              OK
Backup/restore:                OK
Upgrade:                       OK
------------- Test with PHP7.4 -------------
Install (subdir):              OK
Backup/restore:                OK
Upgrade:                       OK
------------- Test with PHP8.0 -------------
Install (subdir):              OK
Backup/restore:                OK
Upgrade:                       OK

Level results
=============
Level 1 (Installable in at least one scenario)    OK
Level 2 (Installable in all scenarios)            OK
Level 3 (Can be upgraded)                         OK
Level 4 (Can be backup/restored)                  OK
Level 5 (No linter errors)                       
Level 6 (App is in a community-operated git org)  ok 
Level 7 (Pass all tests + no linter warnings)    
Level 8 (Maintained and long-term good quality)  
Level 9 (Flagged high-quality in app catalog)    

Global level for this application: 4 (Can be backup/restored)

(Protip™ for CI admin: you should 'apt install wkhtmltopdf optipng --no-install-recommends' to enable result summary export to .png)
Global working time for all tests: 40 minutes, 50 seconds (21:04:35)
```

**Comment by fflorent** [comment]
!testme

**Comment by fflorent** [comment]
!testme

**Comment by fflorent** [comment]
Do you feel comfortable with the changes? I can understand that you would rather go with changing directly the core for that.

**Comment by fflorent** [comment]
Propose to first focus on #107 and then review this.

**Comment by ericgaspar** [comment]
Closing with #116

**Comment by fflorent** [comment]
!testme

8 mai 2023 14:51:57 Éric Gaspar ***@***.***>:

**Pull request closed without merging**